File: LeetCode/Easy/2325-decode-the-message/2325-decode-the-message.java
class Solution {
    public String decodeMessage(String key, String message) {
        int n = message.length();
        char[] decode = new char[26];
        int idx = 0;
        
        char[] keyArray = key.toCharArray();
        char[] messageArray = message.toCharArray();
        
        StringBuilder ret = new StringBuilder();
        
        for (char v : keyArray)
            if (v != ' ' && decode[v-97] == 0)
                decode[v-97] = (char) (97 + idx++);
            
        
        for (char v : messageArray)
            if (v != ' ') ret.append(decode[v-97]);
            else ret.append(' ');
        
        return ret.toString();
    }
}

File: LeetCode/Easy/2325-decode-the-message/2325-decode-the-message.cpp
class Solution {
public:
    string decodeMessage(string key, string message) {
        int n = message.size();
        vector<char> decode(26);
        int idx = 0;
        string ret = "";
        
        for (char v : key)
            if (v != ' ' && decode[v-97] == 0)
                decode[v-97] = (char) (97 + idx++);
            
        
        for (char v : message)
            if (v != ' ') ret += decode[v-97];
            else ret += ' ';
        
        return ret;
    }
};